xla_native_functions.yaml

```diff
@@ -232,6 +232,7 @@ supported:
   - cholesky
   - qr
   - erfinv
+  - sgn
```

Comment:
fyi you should lower `sgn.out` instead. Check [here](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml#L291) and you will see that `sgn` will call `sgn.out`(leaf node) when not being lowered but not vice versa.

Reply:
after chatting with @bdhirsh I take it back, I think we can just lower `sgn`. `sgn_out` will be code-gened from the pt/xla side.

Reply:
and fyi this is only true because of the new codegen 😃

Reply:
lol. I've already implemented it. Should I revert my changes? (they are not pushed yet)

@bdhirsh, to understand this better, what in the new codegen helps support `sgn_out` via `sgn`?

Reply:
@miladm Sorry, please revert. If I understand it correctly
1. `sgn.out` is the leaf node. If we do not lower the `sgn`, from the pytorch level, `sgn` will use `sgn.out` implementation
2. There is also another code gen at pytorch/xla level which will genreate `sgn_out` based on `sgn` implementation. It is recommended(faster?) to implement the `sgn` direcly and let pt/xla code gen to handle the `sgn_out`.

Reply:
`pytorch/xla` calls into an [external codegen API](https://github.com/pytorch/pytorch/blob/f4f2c1231a4a429182d12b58e16b4a42fcae68a5/tools/codegen/gen_backend_stubs.py#L161) in core that knows how to group together `functional`/`inplace`/`out=` variants of the same operator. If it sees that you've provided a lowering for `sgn` but not `sgn.out` (specified through `xla_native_functions.yaml`), it will generate a lowering for `sgn.out` for you, which just calls your functional lowering and sets the IR of the `out=` parameter appropriately. You can see the generated kernel in `torch_xla/csrc/RegisterXLA.cpp` (after building from source).  If you're curious, the logic that generates that code lives [here](https://github.com/pytorch/pytorch/blob/f4f2c1231a4a429182d12b58e16b4a42fcae68a5/tools/codegen/dest/register_dispatch_key.py#L107).

Reply:
@JackCaoG sounds good and no worries at all. I reverted the change.
@bdhirsh @JackCaoG thanks for the clarifications.